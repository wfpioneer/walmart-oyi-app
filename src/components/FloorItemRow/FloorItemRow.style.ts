import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'flex-start',
    backgroundColor: COLOR.WHITE,
    paddingLeft: 15,
    paddingRight: 25,
    paddingVertical: 5,
    marginBottom: 2,
    flexDirection: 'row'
  },
  content: {
    alignItems: 'flex-start',
    justifyContent: 'space-evenly',
    flex: 1
  },
  itemNbr: {
    color: COLOR.GREY_500,
    fontSize: 12,
    lineHeight: 22,
    flex: 1
  },
  itemDesc: {
    lineHeight: 22
  },
  price: {
    color: COLOR.GREY_500,
    lineHeight: 22
  },
  pallet: {
    flexDirection: 'row'
  },
  rightButton: {
    paddingVertical: 3
  },
  leftButton: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  image: {
    width: 25,
    height: 20,
    backgroundColor: COLOR.WHITE
  }
});
