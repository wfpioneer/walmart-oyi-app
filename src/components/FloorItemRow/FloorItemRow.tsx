import React from 'react';
import { Text, View } from 'react-native';
import styles from './FloorItemRow.style';
import { currencies, strings } from '../../locales';
import { SectionDetailsItem } from '../../models/LocationItems';

export type FloorItemRowProps = { item: SectionDetailsItem };

const FloorItemRow = (props: FloorItemRowProps): JSX.Element => {
  const { item } = props;

  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <Text style={styles.itemNbr}>
          {`${strings('ITEM.ITEM')} ${item.itemNbr}`}
        </Text>
        <Text style={styles.itemDesc}>{item.itemDesc}</Text>
        <Text style={styles.price}>{currencies(item.price)}</Text>
      </View>
    </View>
  );
};

export default FloorItemRow;
