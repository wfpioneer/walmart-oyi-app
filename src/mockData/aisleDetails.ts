import { AisleItem } from '../models/LocationItem';

export const mockAisles: AisleItem[] = [
  {
    aisleId: 1,
    aisleName: 'Aisle 1',
    sectionCount: 30
  },
  {
    aisleId: 2,
    aisleName: 'Aisle 2',
    sectionCount: 0
  },
  {
    aisleId: 3,
    aisleName: 'Aisle 3',
    sectionCount: 2
  },
  {
    aisleId: 4,
    aisleName: 'Aisle 4',
    sectionCount: 2
  },
  {
    aisleId: 5,
    aisleName: 'Aisle 5',
    sectionCount: 30
  },
  {
    aisleId: 6,
    aisleName: 'Aisle 6',
    sectionCount: 0
  }
];
