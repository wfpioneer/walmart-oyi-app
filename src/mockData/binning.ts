export const mockPallets = [{
  id: 123456,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '12343534',
    itemNbr: 351231,
    quantity: 2
  }
}, {
  id: 345345,
  expirationDate: '10/3/2022',
  firstItem: {
    itemDesc: 'itemDesc',
    price: 123,
    upcNbr: '345345',
    itemNbr: 351231,
    quantity: 2
  }
}];
