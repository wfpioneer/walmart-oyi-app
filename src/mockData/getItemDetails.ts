import ItemDetails from '../models/ItemDetails';
/* eslint-disable quote-props */

const floorType = 'Sales Floor';
const salesLastUpdateTs = '2020-07-15T08:02:17-05:00';
const salesDay1 = '2020-07-08';
const salesDay2 = '2020-07-09';
const salesDay3 = '2020-07-10';
const salesDay4 = '2020-07-11';
const salesDay5 = '2020-07-12';
const salesDay6 = '2020-07-13';
const salesDay7 = '2020-07-14';

export default {
  '123': {
    itemName: 'Test Item That is Really, Really Long (and has parenthesis)',
    itemNbr: 1234567890,
    upcNbr: '000055559999',
    status: 'Active',
    categoryNbr: 93,
    categoryDesc: 'Meat PI',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    pendingOnHandsQty: -999,
    consolidatedOnHandQty: 10,
    claimsOnHandQty: 5,
    backroomQty: 6,
    cloudQty: undefined,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 100
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 39
        },
        {
          week: 52,
          value: 26
        },
        {
          week: 53,
          value: 45
        },
        {
          week: 54,
          value: 34
        },
        {
          week: 55,
          value: 34
        }
      ]
    }
  } as ItemDetails,
  '456': {
    itemName: 'Small, Store Use Item',
    itemNbr: 987654321,
    upcNbr: '777555333',
    status: 'Active',
    categoryNbr: 99,
    categoryDesc: 'Store Use',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    pendingOnHandsQty: -999,
    consolidatedOnHandQty: 0,
    claimsOnHandQty: 5,
    backroomQty: 6,
    cloudQty: 42,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 10
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 0
        },
        {
          week: 1,
          value: 10
        },
        {
          week: 2,
          value: 27
        },
        {
          week: 3,
          value: 10
        }
      ]
    }
  } as ItemDetails,
  '789': {
    itemName: 'Wine Item',
    itemNbr: 987654321,
    upcNbr: '777555333',
    status: 'Active',
    categoryNbr: 19,
    categoryDesc: 'Wine',
    price: 2000.94,
    basePrice: 1500.30,
    exceptionType: 'nsfl',
    imageUrlKey: undefined,
    imageBlobKey: undefined,
    completed: false,
    onHandsQty: 42,
    consolidatedOnHandQty: 0,
    claimsOnHandQty: 0,
    backroomQty: 6,
    pendingOnHandsQty: 35,
    cloudQty: 42,
    replenishment: {
      onOrder: 48
    },
    location: {
      floor: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: floorType,
          typeNbr: 8
        },
        {
          zoneId: 0,
          aisleId: 0,
          sectionId: 2,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '2',
          locationName: 'A1-2',
          type: 'End Cap',
          typeNbr: 12
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 3,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '3',
          locationName: 'A1-3',
          type: 'Pod',
          typeNbr: 13
        },
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 4,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '4',
          locationName: 'A1-4',
          type: 'Display',
          typeNbr: 11
        }
      ],
      reserve: [
        {
          zoneId: 0,
          aisleId: 1,
          sectionId: 1,
          zoneName: 'A',
          aisleName: '1',
          sectionName: '1',
          locationName: 'A1-1',
          type: 'Reserve',
          typeNbr: 7
        }
      ],
      count: 5
    },
    sales: {
      lastUpdateTs: salesLastUpdateTs,
      dailyAvgSales: 15,
      daily: [
        {
          day: salesDay1,
          value: 10
        },
        {
          day: salesDay2,
          value: 0
        },
        {
          day: salesDay3,
          value: 10
        },
        {
          day: salesDay4,
          value: 10
        },
        {
          day: salesDay5,
          value: 42
        },
        {
          day: salesDay6,
          value: 5
        },
        {
          day: salesDay7,
          value: 1
        }
      ],
      weeklyAvgSales: 10,
      weekly: [
        {
          week: 51,
          value: 0
        },
        {
          week: 1,
          value: 10
        },
        {
          week: 2,
          value: 27
        },
        {
          week: 3,
          value: 10
        }
      ]
    }
  } as ItemDetails
};
