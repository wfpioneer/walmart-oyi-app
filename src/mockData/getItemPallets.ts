import { AsyncState } from '../models/AsyncState';

export const itemPallets = {
  itemNbr: 987654321,
  upcNbr: '777555333',
  category: 99,
  pallets: [{
    palletId: '4598',
    quantity: 22,
    sectionId: 5578,
    locationName: 'D1-4',
    mixedPallet: false
  }]
};

export const mockGetItemPalletsAsyncState : AsyncState = {
  isWaiting: false,
  value: null,
  result: {
    status: 200,
    data: itemPallets
  },
  error: null
};
