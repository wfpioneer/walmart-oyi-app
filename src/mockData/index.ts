
import itemDetails from './getItemDetails';


export const getMockItemDetails = (upc: string) => {
  // @ts-ignore
  return itemDetails[upc] || {};
};
