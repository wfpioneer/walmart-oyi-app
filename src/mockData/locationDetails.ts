import { LocationItem } from '../models/LocationItems';

export const mockLocationDetails : LocationItem = {
  zone: {
    id: 1,
    name: 'G - Grocery'
  },
  aisle: {
    id: 1,
    name: 'Aisle 1'
  },
  section: {
    id: 1,
    name: 'Section 1'
  },
  floor: [
    {
      itemNbr: 980078597,
      itemDesc: 'Nature Valley Sweet and Salty Granola Bars',
      price: 2.40
    },
    {
      itemNbr: 980011111,
      itemDesc: 'Nature Valley Honey Bars',
      price: 5.80
    }

  ],
  reserve: [
    {
      palletId: 1,
      palletCreateTS: '2021-1-1',
      items: [
        {
          itemNbr: 1,
          itemDesc: 'Cabbage',
          price: 2.40
        }
      ]
    }
  ]
};

export const mockLocationDetailsEmpty : LocationItem = {
  ...mockLocationDetails,
  floor: [],
  reserve: []
};

export const mockLocationDetailsLargeLocationCount : LocationItem = {
  ...mockLocationDetails,
  floor: Array.from({ length: 100 }, () => mockLocationDetails.floor[0]),
  reserve: Array.from({ length: 100 }, () => mockLocationDetails.reserve[0])
};
