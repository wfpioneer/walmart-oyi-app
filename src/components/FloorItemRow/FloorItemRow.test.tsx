import { NavigationProp } from '@react-navigation/native';
import React from 'react';
import { Provider } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import store from '../../state';
import { mockLocationDetails } from '../../mockData/locationDetails';
import FloorItemRow from './FloorItemRow';

const mockFloorItem = mockLocationDetails.items.sectionItems[0];
let navigationProp: NavigationProp<any>;

describe('FloorItemRow Component', () => {
  it('Renders a FloorItem', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <Provider store={store}>
        <FloorItemRow item={mockFloorItem} dispatch={jest.fn()} navigation={navigationProp} />
      </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders a FloorItem with a long name', () => {
    const longItemDesc = new Array(5).fill('An Item With A Very Long Item Description').join(' ');
    const mockFloorItemLongName = { ...mockFloorItem, itemDesc: longItemDesc };
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <Provider store={store}>
        <FloorItemRow item={mockFloorItemLongName} dispatch={jest.fn()} navigation={navigationProp} />
      </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
