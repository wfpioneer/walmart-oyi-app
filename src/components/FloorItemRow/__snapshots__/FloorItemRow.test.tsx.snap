// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloorItemRow Component Renders a FloorItem 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "identityFunctionCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <FloorItemRow
    dispatch={[MockFunction]}
    item={
      {
        "itemDesc": "Nature Valley Sweet and Salty Granola Bars",
        "itemNbr": 980078597,
        "locationType": 8,
        "price": 2.4,
        "upcNbr": "123456789098",
      }
    }
    tokenIsWaiting={false}
    trackEventCall={[MockFunction]}
  />
</Context.Provider>
`;

exports[`FloorItemRow Component Renders a FloorItem with a long name 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "identityFunctionCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <FloorItemRow
    dispatch={[MockFunction]}
    item={
      {
        "itemDesc": "An Item With A Very Long Item Description An Item With A Very Long Item Description An Item With A Very Long Item Description An Item With A Very Long Item Description An Item With A Very Long Item Description",
        "itemNbr": 980078597,
        "locationType": 8,
        "price": 2.4,
        "upcNbr": "123456789098",
      }
    }
    tokenIsWaiting={false}
    trackEventCall={[MockFunction]}
  />
</Context.Provider>
`;
